### DISCUSSION TITLE
xDrip with G7

### DISCUSSION THREAD

**Discussion by Jared-Is-Coding** [NightscoutFoundation/xDrip#3367] [Q&A]
Hey all,

I'm on a Pixel 7a, using xDrip nightly from February 14th. After a relatively normal sensor session, I started receiving "Mismatch" errors in the system status for the collector. I attempted:
- Restart bluetooth
- Restart xDrip
- Restart phone
- Reinstall xDrip in place
- Update xDrip to latest nightly

All while definitely out of range of any other running sensors. No luck. Due to time constraints for the day, I just replaced it.

I wanted to post and (1) ask about this particular issue but also (2) just get any suggestions for G7 on xDrip in general. Using the latest nightly now, just started a new session.

1. I can't seem to find a stop sensor button. Does it exist? If I need to stop a session early due to removal for various reasons, how do I go about this other than just forgetting the sensor in my bluetooth settings and removing my "Sensor Code" setting in the menu?
2. Any concerns with unopened sensors in their boxes being nearby when doing startups? My assumption was that they're "asleep" until the applicator places them. That sound right?

As for the xDrip project, I'm happy to provide logs or feedback as needed related to G7 with xDrip as the collector.

Thanks all!

**Comment by Navid200** [comment]
To avoid a disconnect to start, you should disable any battery optimization on the phone.  Most can disable xDrip because it is sleep most of the time.
I have  Pixel 6a and all I have to do in that respect is to set the xDrip battery setting under Android settings to unrestricted.
But, it is possible that Google has made 7a more restrictive.  So, there may be other settings you need to disable.

After a disconnect problem has started, to reconnect, it may help to remove any previously connected device from the list including the one related to the current device you want to reconnect to.
But, reconnecting is not going to resolve the cause of the disconnect.

There is never a reason to stop a G7.  So, I removed the option.  If you are done with a G7, just peel it off.

Unopened devices cannot interfere.

**Reply by Jared-Is-Coding** [reply]
Thanks @Navid200,

Re: battery - Battery usage has been set to "Unrestricted" since installation, all set there!

Re: reconnect - By "remove ... from the list" are you referring to the phone system bluetooth device connection list? Sure. Definitely want to monitor this kind of thing, as I wouldn't want sensors randomly failing like this in the future. Any kind of logs I can poke through ~~or send your way~~ (edit: Confused you with Jam, oops)?

Re: stopping - fair enough. I might like the option for the peace of mind, regardless.

Thanks again.

**Reply by Navid200** [reply]
Yes, use Android to forget all Bluetooth devices starting with DXCM.

Please explain what you mean by wanting the option for peace of mind.

There are users who stop a Libre session and start again on a regular basis.
They would switch to G7 and think it was the same and stop and then realize that they have to throw it away now.
I didn't want that to happen.  So, I removed the stop option.

I am a G7 user.  There is no time I need to stop a G7.  Even after the session ends, G7 continues to transmit for about 90 more days.  There is no change in that behavior whether you stop it or not.
Please explain what you are saying.  Can you think of a reason why one would need to stop a G7?

**Reply by Jared-Is-Coding** [reply]
@Navid200 

Peace of mind as it relates to what I saw happening today - Want to make sure the removed transmitter won't interfere with a new one being started up.

However, based on what you're saying it's no different whether it's stopped or not? If it just continues transmitting, then I agree that no button would ever be needed.

**Reply by Jared-Is-Coding** [reply]
As a related note, I'm not seeing new G7 sensor sessions log a "start new sensor" event. Is this intended?

The collector screen even shows the age of the previous sensor rather than the latest one.

**Reply by Navid200** [reply]
There is an open issue related to the start sensor time reported on the classic status page being wrong for G7:
https://github.com/NightscoutFoundation/xDrip/issues/3250

And I have an open PR that fixes it: https://github.com/NightscoutFoundation/xDrip/pull/3263

**Comment by Jared-Is-Coding** [comment]
Going to mark this as closed, since it was a one-off event so far and a PR exists for the unrelated issue. Will log further discussion if necessary.
